virtual_rainforest/models/abiotic/radiation.py

```diff
@@ -0,0 +1,208 @@
+                and 1 (cloud free sky)
+            albedo_shortwave: shortwave albedo, default = 0.17
+        """
+        self.topofcanopy_radiation = (1.0 - albedo_shortwave) * tau * shortwave_in
+
+    def calc_longwave_radiation(
+        self,
+        canopy_temperature: NDArray[np.float32],
+        surface_temperature: NDArray[np.float32],
+    ) -> None:
+        """Calculates longwave emission from canopy and forest floor [J m-2].
+
+        Args:
+            canopy_temperature: canopy temperature of n layers [C]; the array size is
+                set to max number of layers (n_max) and filled with NaN where n < n_max
+            surface_temperature: surface soil temperature [C]
+        """
+        # longwave emission canopy
+        self.longwave_canopy = (
+            CANOPY_EMISSIVITY
+            * BOLZMAN_CONSTANT
+            * (CELSIUS_TO_KELVIN + canopy_temperature) ** 4
+        )
+
+        # longwave emission surface
+        self.longwave_soil = (
+            SOIL_EMISSIVITY
+            * BOLZMAN_CONSTANT
+            * (CELSIUS_TO_KELVIN + surface_temperature) ** 4
+        )
+
+    def calc_netradiation_surface(
+        self,
+        canopy_absorption: NDArray[np.float32],
+    ) -> None:
+        """Calculates daily net radiation at the forest floor [J m-2].
+
+        Args:
+            canopy_absorption: absorption by canopy layers, [J m-2]
+        """
+        self.netradiation_surface = (
+            self.topofcanopy_radiation
+            - canopy_absorption  # np.sum(canopy_absorption, axis=1)  # what axis?
+            - self.longwave_soil
+            - self.longwave_canopy  # np.sum(self.longwave_canopy, axis=1)
+        )
+
+    def radiation_balance(
+        self,
+        shortwave_in: NDArray[np.float32],
+        sunshine_fraction: NDArray[np.float32],
+        albedo_vis: NDArray[np.float32],
+        albedo_shortwave: NDArray[np.float32],
+        canopy_temperature: NDArray[np.float32],
+        surface_temperature: NDArray[np.float32],
+        canopy_absorption: NDArray[np.float32],
+    ) -> None:
+        """Calculate radiation balance.
+
+        The function runs the full radiation balance and populates the following
+        attributes:
+            * elevation [m]
+            * top of canopy downward shortwave radiation, [J m-2]
+            * top of canopy photosynthetic photon flux density,[mol m-2]
+            * longwave radiation from n individual canopy layers [J m-2]
+            * longwave radiation from soil [J m-2]
+            * net shortwave radiation at the surface (=forest floor)[J m-2]
+
+        Args:
+            elevation: elevation [m]
+            shortwave_in: daily downward shortwave radiation[J m-2]
+            sunshine_fraction: fraction of sunshine hours, between 0
+                (100% cloud cover) and 1 (cloud free sky)
+            albedo_vis: visible light albedo, default = 0.03
+            albedo_shortwave: shortwave albedo, default = 0.17
```

Comment:
If you indicated in the docstring a default value, then that default value should be included in the signature. For that to work, you will need to add these two to the end of the list of input arguments.

Reply:
Thanks for pointing this out, I think it goes back to the problem of constants discussed earlier. I hope this can be solved with the restructuring of the module